src/backend/api/meals.js

```diff
@@ -1,15 +1,152 @@
+const { query } = require("express");
 const express = require("express");
+const { max } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
+router.get("/", async (req, res) => {
+
+  if ("maxPrice" in req.query) {
+    const maxPrice = Number(req.query.maxPrice);
+    if (!maxPrice) {
+      res.status(403).json('Please Enter Max Prise of Meal')
+    } else {
+      const meals = await knex("meal").where("price", "<", maxPrice);
+      res.json(meals)
```

Comment:
Did you think about the scenario where there is a `maxPrice` and also a `title` parameter? In that case you would respond here but later down in the code you (on line 43) you would try to send another response and that would fail.